copr-backend FTBFS in Rawhide

There are many test-failures like this:

```
_________________________ test_full_rpm_build_no_sign __________________________

_parse_results = <MagicMock name='_parse_results' id='140330113770864'>
f_build_rpm_case = Munch({'workdir': '/tmp/build-worker-tests-sraek8gu', 'be_config_file': '/tmp/build-worker-tests-sraek8gu/copr-be.conf...gicMock name='ResallocHostFactory' id='140330113432608'>, 'ssh': <testlib.FakeSSHConnection object at 0x7fa126950740>})
caplog = <_pytest.logging.LogCaptureFixture object at 0x7fa126952420>

    @_patch_bwbuild_object("BuildBackgroundWorker._parse_results")
    def test_full_rpm_build_no_sign(_parse_results, f_build_rpm_case, caplog):
        """
        Go through the whole (successful) build of a binary RPM
        """
        worker = f_build_rpm_case.bw
        worker.process()
    
        results = worker.job.results_dir
        assert os.path.exists(os.path.join(results, "builder-live.log.gz"))
        assert os.path.exists(os.path.join(results, "backend.log.gz"))
    
        found_success_log_entry = False
        for record in caplog.record_tuples:
            _, level, msg = record
            assert level < logging.ERROR
            if "Finished build: id=848963 failed=False" in msg:
                found_success_log_entry = True
        assert found_success_log_entry
    
>       repodata = load_primary_xml(os.path.join(results, "..", "repodata"))

tests/test_background_worker_build.py:294: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/testlib/repodata.py:20: in load_primary_xml
    xml_content = fprimary.read()
/usr/lib64/python3.12/gzip.py:324: in read
    return self._buffer.read(size)
/usr/lib64/python3.12/_compression.py:118: in readall
    while data := self.read(sys.maxsize):
/usr/lib64/python3.12/gzip.py:524: in read
    if not self._read_gzip_header():
/usr/lib64/python3.12/gzip.py:493: in _read_gzip_header
    last_mtime = _read_gzip_header(self._fp)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

fp = <gzip._PaddedFile object at 0x7fa126c3b4a0>

    def _read_gzip_header(fp):
        '''Read a gzip header from `fp` and progress to the end of the header.
    
        Returns last mtime if header was present or None otherwise.
        '''
        magic = fp.read(2)
        if magic == b'':
            return None
    
        if magic != b'\037\213':
>           raise BadGzipFile('Not a gzipped file (%r)' % magic)
E           gzip.BadGzipFile: Not a gzipped file (b'(\xb5')
```

--- a/backend/copr-backend.spec
+++ b/backend/copr-backend.spec
@@ -56,6 +56,7 @@ BuildRequires: python3-retask
 BuildRequires: python3-setproctitle
 BuildRequires: python3-sphinx
 BuildRequires: python3-tabulate
+BuildRequires: python3-zstandard
 BuildRequires: modulemd-tools >= 0.6
 BuildRequires: prunerepo >= %prunerepo_version
 BuildRequires: dnf

--- a/backend/tests/test_modifyrepo.py
+++ b/backend/tests/test_modifyrepo.py
@@ -547,9 +547,10 @@ class TestModifyRepo(object):
             pass
 
         assert call_copr_repo(chrootdir)
-        name = glob.glob(os.path.join(chrootdir, "repodata", "*-comps.xml.gz"))
-        assert os.path.exists(name[0])
 
+        name = glob.glob(os.path.join(chrootdir, "repodata", "*-comps.xml.*"))
+        assert os.path.exists(name[0])
+        assert name[0].endswith((".zst", ".gz"))
 
     @skip("Works locally and in Copr but fails in Koji")
     @mock.patch("copr_prune_results.LOG", logging.getLogger())

--- a/backend/tests/testlib/repodata.py
+++ b/backend/tests/testlib/repodata.py
@@ -7,6 +7,14 @@ import glob
 import gzip
 from xml.dom import minidom
 
+try:
+    # This import tracebacks on F37 but we need it only on F39+
+    # pylint: disable=import-outside-toplevel
+    from zstandard import ZstdDecompressor
+except ImportError:
+    pass
+
+
 def load_primary_xml(dirname):
     '''
     Parse priary.xml from the given repodata directory path, and return
@@ -16,8 +24,7 @@ def load_primary_xml(dirname):
     hrefs = set()
     names = set()
     primary = glob.glob(os.path.join(dirname, '*primary*xml*'))[0]
-    with gzip.open(primary) as fprimary:
-        xml_content = fprimary.read()
+    xml_content = extract(primary)
 
     dom = minidom.parseString(xml_content)
 
@@ -36,3 +43,14 @@ def load_primary_xml(dirname):
         'hrefs': hrefs,
         'names': names,
     }
+
+def extract(path):
+    if path.endswith(".zst"):
+        with open(path, "rb") as fp:
+            decompressor = ZstdDecompressor()
+            return decompressor.stream_reader(fp).read()
+
+    if path.endswith(".gz"):
+        with gzip.open(path) as fp:
+            return fp.read()
+    raise ValueError("Unexpected extension: {0}".format(path))

